libcodechecker/generic_package_context.py

```diff
@@ -23,6 +24,36 @@
 LOG = logger.get_logger('system')
 
 
+class SeverityMap(MutableMapping):
```

Comment:
If we do not want to change the values of the severity levels after initialization a `Mapping` would be enough too,  `__setitem__` and `__delitem__` can be removed in that case.
Maybe it would be better so the severity levels can not be changed run time.